master/internal/webhooks/api_webhook.go

```diff
@@ -223,3 +230,29 @@ func (a *WebhooksAPIServer) TestWebhook(
 	}
 	return &apiv1.TestWebhookResponse{}, nil
 }
+
+// PostWebhookEventData handles data for custom trigger.
+func (a *WebhooksAPIServer) PostWebhookEventData(
```

Comment:
Can we maybe add a couple of integration tests for this function?

Reply:
e2e tests for this function were temporarily dropped during a merge, now they should be restored